phpstan.neon

```diff
@@ -1,7 +1,23 @@
 parameters:
+    level: 7
+
+    paths:
+        - lib
+        - tests
+
     autoload_files:
         - %currentWorkingDirectory%/tests/Doctrine/Tests/Persistence/ObjectManagerDecoratorTest.php
         - %currentWorkingDirectory%/tests/Doctrine/Tests/Persistence/Mapping/ClassMetadataFactoryTest.php
         - %currentWorkingDirectory%/tests/Doctrine/Tests/Persistence/Mapping/_files/annotation/TestClass.php
+
     excludes_analyse:
         - %currentWorkingDirectory%/tests/Doctrine/Tests/Persistence/Mapping/_files/TestEntity.php
+
+    ignoreErrors:
+        # looks like a bug in phpstan since method_exists first argument accepts a string
+        - '#Call to function method_exists\(\) with string and (.*) will always evaluate to false#'
```

Comment:
Where does this error appear? I believe PHPStan pretty much does the `method_exists` call you add to detect any useless calls. This may be an error in type hinting or type documentation.

Reply:
It smells like a bug in phpstan to me. `method_exists` accepts a class name or object instance and PHPStan seems to think it can only accept an object instance. https://github.com/doctrine/persistence/blob/types/lib/Doctrine/Persistence/Mapping/Driver/StaticPHPDriver.php#L134

Reply:
Yes, [it's a bug](https://github.com/phpstan/phpstan/issues/1482)